sklearn/base.py

```diff
@@ -202,12 +202,8 @@ def get_params(self, deep=True):
             try:
                 value = getattr(self, key)
             except AttributeError:
-                warnings.warn('From version 0.24, get_params will raise an '
-                              'AttributeError if a parameter cannot be '
-                              'retrieved as an instance attribute. Previously '
-                              'it would return None.',
-                              FutureWarning)
-                value = None
+                raise AttributeError('The parameter {} cannot be retrieved '
```

Comment:
Would it be better to just remove the try-except?

Reply:
Removing the `try`-`except` will fail (`AttributeError`) for those parameters given in `__init__` but not stored.